packages/api/cms-api/src/blocks/block.ts

```diff
@@ -69,7 +73,7 @@ export interface BlockDataInterface {
     transformToPlain(context: BlockContext): Promise<Type<BlockTransformerServiceInterface> | TraversableTransformBlockResponse>;
     transformToSave(): TraversableTransformBlockResponse;
     indexData(): BlockIndexData;
-    warnings(): BlockWarning[];
+    warnings(): BlockWarning[] | Promise<BlockWarning[]> | Type<BlockWarningsServiceInterface>;
```

Comment:
Is there a use case for `Promise<BlockWarning[]>`? We probably can't do any async code without using a service with dependency injection.

Reply:
I could imagine that an external link might be a use case. A fetch to check if the link is available.

Reply:
I can also remove the Promise, because I can't think of any other use case at the moment.

Reply:
Didn't think of that, that's a valid use case. Let's keep it